src/main/java/com/redhat/devtools/intellij/common/listener/FileEditorListener.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.intellij.common.listener;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.FileEditorManagerListener;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static com.redhat.devtools.intellij.common.CommonConstants.TEKTON_RS;
+
+public class FileEditorListener implements FileEditorManagerListener {
```

Comment:
Either we keep it Tekton independant and remove the TEKTON constants and rename it to DeleteOnCloseListener. Another better solution would be to create the file in temporaty folder and then set the deleteOnExit flag so we don't have to get some specific code to handle.